src/endpoints/mod_feedback.rs

```diff
@@ -0,0 +1,158 @@
+use actix_web::{get, post, delete, web, HttpResponse, Responder};
+use serde::{Deserialize};
+use sqlx::Acquire;
+
+use crate::{
+    extractors::auth::Auth,
+    types::{
+        api::{ApiError, ApiResponse},
+        models::{
+            developer::{Developer},
+        },
+    },
+    AppData
+};
+use crate::types::models::mod_version::ModVersion;
+use crate::types::models::mod_feedback::{ModFeedback,FeedbackTypeEnum};
+use crate::types::models::mod_version_status::ModVersionStatusEnum;
+
+#[derive(Deserialize)]
+pub struct GetModFeedbackPath {
+    id: String,
+    version: String
+}
+
+#[derive(Deserialize)]
+pub struct PostModFeedbackPayload {
+    feedback_type: FeedbackTypeEnum,
+    feedback: String,
+}
+
+#[get("/v1/mods/{id}/versions/{version}/feedback")]
+pub async fn get_mod_feedback(
+    data: web::Data<AppData>,
+    path: web::Path<GetModFeedbackPath>,
+    auth: Auth,
+) -> Result<impl Responder, ApiError> {
+    let dev = auth.developer()?;
+    let mut pool = data.db.acquire().await.or(Err(ApiError::DbAcquireError))?;
+
+    let access = Developer::has_access_to_mod(dev.id, &path.id, &mut pool).await?;
+
+    if !access && !dev.admin && !dev.verified {
+        return Err(ApiError::Forbidden);
+    }
+
+    let mut note_only = false;
+    if !access && !dev.admin {
+        note_only = true;
```

Comment:
i'm not sure what this is for? think it would be fine if verified devs could view the suggestions that other devs (or themselves) make

Reply:
it would lead to a peer bias - i was considering it but i want the decisions to be independent

Reply:
ah yeah, i understand now
guess my current feedback is that this doesn't have to be mutable :p 
you can just `let note_only = !access && !dev.admin`

Reply:
ok, but i'll also change note_only to mod_dev_only - once again, to remove peer bias